lib/src/components/NxCollapsibleItemsSelect/NxCollapsibleMultiSelect/__tests__/NxCollapsibleMultiSelect.test.tsx

```diff
@@ -4,355 +4,559 @@
+      expect(renderAndGetTrigger({ isOpen: true })).toHaveAttribute('aria-expanded', 'true');
+    });
+
+    it('sets disabled if the disabled prop is true or there are no options', function() {
+      expect(renderAndGetTrigger()).toBeEnabled();
+      expect(renderAndGetTrigger({ disabled: true })).toBeDisabled();
+      expect(renderAndGetTrigger({ disabled: null })).toBeEnabled();
+      expect(renderAndGetTrigger({ disabled: undefined })).toBeEnabled();
 
-      expect(renderOption({id: 'foo', name: 'Foo'})).toHaveProp('isChecked', false);
-      expect(renderOption({id: 'bar', name: 'Bar'})).toHaveProp('isChecked', false);
-      expect(renderOption({id: null, name: 'Null'})).toHaveProp('isChecked', false);
+      expect(renderAndGetTrigger({ options: [] })).toBeDisabled();
+      expect(renderAndGetTrigger({ options: [], disabled: true })).toBeDisabled();
+      expect(renderAndGetTrigger({ options: [], disabled: null })).toBeDisabled();
+      expect(renderAndGetTrigger({ options: [], disabled: undefined })).toBeDisabled();
     });
 
-    it('renders enabled <NxCheckbox> for all options if component is not disabled', function () {
-      const renderOption = getShallow().renderProp('renderOption');
+    it('fires the component\'s onToggleCollapse when clicked if there are options', async function() {
+      const user = userEvent.setup(),
+          onToggleCollapseOptions = jest.fn(),
+          onToggleCollapseNoOptions = jest.fn(),
+          optionsTrigger = renderAndGetTrigger({
+            onToggleCollapse: onToggleCollapseOptions
+          }),
+          noOptionsTrigger = renderAndGetTrigger({
+            onToggleCollapse: onToggleCollapseNoOptions,
+            options: []
+          });
+
+      expect(onToggleCollapseNoOptions).not.toHaveBeenCalled();
+      await user.click(noOptionsTrigger);
+      expect(onToggleCollapseNoOptions).not.toHaveBeenCalled();
 
-      expect(renderOption({id: 'foo', name: 'Foo'})).toHaveProp('disabled', false);
-      expect(renderOption({id: 'bar', name: 'Bar'})).toHaveProp('disabled', false);
-      expect(renderOption({id: null, name: 'Null'})).toHaveProp('disabled', false);
+      expect(onToggleCollapseOptions).not.toHaveBeenCalled();
+      await user.click(optionsTrigger);
+      expect(onToggleCollapseOptions).toHaveBeenCalled();
     });
 
-    it('renders disabled <NxCheckbox> for all options if component is disabled', function () {
-      const shallowRender = getShallow({
-        disabled: true
+    describe('tooltip', function() {
+      describe('when component is disabled due to no options', function() {
+        it('has a default text constructed with the name prop', async function() {
+          const trigger = renderAndGetTrigger({ options: [] });
+
+          await runTimers();
+          expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+          fireEvent.mouseOver(trigger);
```

Comment:
`userEvent` has a `hover` function that would be preferred

Reply:
> userEvent has a hover function that would be preferred

I originally had the `userEvent` for hovering on the disabled trigger in this test but was not able to query the tooltip after waiting `runTimers`.
 
It seems the user-event `hover` does not trigger `mouseover` for disabled elements.
https://github.com/testing-library/user-event/issues/715

Reply:
> I originally had the userEvent for hovering on the disabled trigger in this test but was not able to query the tooltip after waiting runTimers.

It seems the user-event hover does not trigger mouseover for disabled elements.
https://github.com/testing-library/user-event/issues/715

Ah. as that thread says, attribute-disabled buttons do not in fact dispatch mouse events.  Our implementation mitigates that by attaching the tooltip to a div which wraps the button.  Even though it would make the test aware of our implementation, I think it would be slightly preferable to adjust this test to user `user.hover()` on the wrapper div, rather than what you have here.  Neither approach is perfect from a test cleanliness perspective, but at least that way the event dispatching matches what happens in real life.

Reply:
Updated here 0940f9ea7961ef49aa45ea3665bad499f7853a60